build-info-extractor-pip/src/main/java/org/jfrog/build/extractor/pip/extractor/PipBuildInfoExtractor.java

```diff
@@ -57,58 +59,126 @@ Build extract(ArtifactoryDependenciesClient client, String repository, String in
+                aqlQueries.add(getPipDependenciesAql(filesQueryPartBuilder, repository));
+                filesCounter = 0;
+                filesQueryPartBuilder.setLength(0);
+            }
+        }
+        if (filesQueryPartBuilder.length() > 0) {
+            aqlQueries.add(getPipDependenciesAql(filesQueryPartBuilder, repository));
+        }
+
+        // Perform queries and create Dependency objects.
+        AqlSearchResult searchResult = runAqlQueries(aqlQueries, client);
+        Map<String, Dependency> dependenciesMap = createDependenciesFromAqlResult(searchResult, fileToPackageMap, logger);
+
+        return dependenciesMap;
+    }
+
+    private String getPipDependenciesAql(StringBuilder filesQueryPartBuilder, String repository) {
+        filesQueryPartBuilder.setLength(filesQueryPartBuilder.length() - 1);
+        return String.format(PIP_AQL_FORMAT, repository, filesQueryPartBuilder.toString());
+    }
 
-        try {
-            searchResult = client.searchArtifactsByAql(aql);
-        } catch (IOException e) {
-            throw new IOException("Failed fetching checksums of file: '" + fileName + "' from Artifactory ", e);
+    private AqlSearchResult runAqlQueries(List<String> aqlQueries, ArtifactoryDependenciesClient client) throws IOException {
+        AqlSearchResult aggregatedResults = new AqlSearchResult();
+        for (String aql : aqlQueries) {
+            try {
+                AqlSearchResult searchResult = client.searchArtifactsByAql(aql);
+                if (!searchResult.getResults().isEmpty()) {
+                    aggregatedResults.getResults().addAll(searchResult.getResults());
+                }
+            } catch (IOException e) {
+                throw new IOException("Failed fetching dependencies checksums from Artifactory ", e);
+            }
         }
+        return aggregatedResults;
+    }
 
+    private Map<String, Dependency> createDependenciesFromAqlResult(AqlSearchResult searchResult, Map<String, String> fileToPackage, Log logger) {
         if (searchResult.getResults().isEmpty()) {
-            logger.debug("File: " + fileName + " could not be found in repository: " + repository);
             return null;
         }
 
-        AqlSearchResult.SearchEntry searchEntry = searchResult.getResults().get(0);
-        if (StringUtils.isBlank(searchEntry.getActualMd5()) || StringUtils.isBlank(searchEntry.getActualSha1())) {
-            logger.debug("Missing checksums for file: " + fileName + ", sha1: '" + searchEntry.getActualSha1() + "' md5: '" + searchEntry.getActualMd5() + "'");
-            return null;
+        Map<String, Dependency> dependenciesMap = new HashMap<>();
+        for (AqlSearchResult.SearchEntry searchEntry : searchResult.getResults()) {
+            if (StringUtils.isBlank(searchEntry.getName()) || StringUtils.isBlank(searchEntry.getActualMd5())
+                    || StringUtils.isBlank(searchEntry.getActualSha1())) {
```

Comment:
We can make the above if statement shorter and more readable, if we create a method named *isResultComplete*, which returns - 
```
StringUtils.isNotBlank(searchEntry.getName()) && StringUtils.isNotBlank(searchEntry.getActualMd5())
                    && StringUtils.isNotBlank(searchEntry.getActualSha1())
```